superset/views/core.py

```diff
@@ -1221,10 +1221,14 @@ def explore(self, datasource_type=None, datasource_id=None):
             )
 
         standalone = request.args.get("standalone") == "true"
+        user_roles = [role.name.lower() for role in get_user_roles()]
+        is_admin_role = "admin" in user_roles
         bootstrap_data = {
             "can_add": slice_add_perm,
             "can_download": slice_download_perm,
             "can_overwrite": slice_overwrite_perm,
+            "can_overwrite_datasource": is_admin_role
+            or datasource.created_by_fk == user_id,
```

Comment:
add: `or datasource.updated_by_fk == user_id`?

Reply:
if current user is not creator (and not a admin role), that's the exactly ones we want to prevent from changing datasource, even he/she might be able to update before.